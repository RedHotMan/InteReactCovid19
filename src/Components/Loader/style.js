import styled, { keyframes } from "styled-components";

const spin = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

export const LoaderDiv = styled.div`
  border: 6px solid #d6d6d6;
  border-top: 6px solid #255fff;
  border-radius: 50%;
  width: 30px;
  height: 30px;
  animation: ${spin} 1s linear infinite;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50m);
`;
