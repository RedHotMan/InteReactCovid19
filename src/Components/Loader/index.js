import React from "react";
import { LoaderDiv } from "./style";

const Loader = () => {
  return <LoaderDiv />;
};

export default Loader;
