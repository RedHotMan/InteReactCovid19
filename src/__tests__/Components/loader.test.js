import React from "react";
import { shallow } from "enzyme";
import Loader from "../../Components/Loader";

describe("Loader", () => {
  it("Should render Loader correctly", () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
