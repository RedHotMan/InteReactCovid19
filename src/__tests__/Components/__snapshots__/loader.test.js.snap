// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Should render Loader correctly 1`] = `"<styled.div />"`;
